client/components/overviewWidget/AddToCart.jsx

```diff
@@ -39,27 +41,69 @@ class AddToCart extends React.Component {
           console.error(err);
         });
     }
+
+    if (this.props.selectedStyle !== prevProps.selectedStyle) {
+      const newSelectedStyleData = this.state.styles.filter((style) => {
+        return style.style_id === parseInt(this.props.selectedStyle);
+      });
+      let skus = Object.values(newSelectedStyleData[0].skus);
+      let sizes = [];
+      skus.forEach((sku) => {
+        sizes.push(sku.size);
+      });
+      this.setState({
+        selectedStyle: newSelectedStyleData[0],
+        sizes
+      });
+      this.sizeSelectorClickHandler({ target: { value: 'select-size' } });
+    }
+  }
+
+  sizeSelectorClickHandler(e) {
+    const selectedSize = e.target.value;
+    if (selectedSize === 'select-size') {
+      this.setState({
+        selectedSizeQuantities: []
+      });
+    } else {
+      const selectedStyle = this.state.selectedStyle;
+      const selectedSku = Object.values(selectedStyle.skus).filter((sku) => {
+        return sku.size === selectedSize;
+      });
+      const selectedSizeQuantity = selectedSku[0].quantity;
+      const maxAvailableQuantity = Math.min(selectedSizeQuantity, 15);
+      const availableQuantities = [];
+      for (let i = 1; i <= maxAvailableQuantity; i++) {
+        availableQuantities.push(i);
+      }
+      this.setState({
+        selectedSizeQuantities: availableQuantities
+      });
+    }
+  }
+
+  addToBagHandler() {
+    console.log('Item added to bag!');
   }
 
   render () {
     return (
       <div>
         <br />
         <form>
-          <select id="select-size" name="select-size">
+          <select id="select-size" name="select-size" onChange={this.sizeSelectorClickHandler}>
```

Comment:
I think right now your controlled react form is missing a piece of that feedback loop.
The select tag I believe should also have a 'value' attribute which is set by the components state.

In this case the tag would be: `<select value={this.state.selectedSize} id="select-size" name="select-size" onChange={this.sizeSelectorClickHandler}>`

You can then use that change of state to trigger other renders that dependent on that value